packages/mui-base/src/RadioGroup/Item/RadioGroupItem.types.ts

```diff
@@ -0,0 +1,29 @@
+import type { BaseUIComponentProps } from '../../utils/types';
+
+export type RadioGroupItemOwnerState = {
+  disabled: boolean;
+  checked: boolean;
+};
+
+export interface RadioGroupItemProps
+  extends Omit<BaseUIComponentProps<'button', RadioGroupItemOwnerState>, 'name'> {
+  /**
+   * The unique identifying value of the radio button in the group.
+   */
+  value: string;
```

Comment:
We don't have to enforce strings here. Let's make it `any` or `unknown` as in Tabs.

Reply:
A hidden `<input>` is rendered for the whole RadioGroup root which requires the native DOM type:

```js
{props.name && (
  <input type="hidden" name={props.name} value={checkedItem ?? ''} required={required} />
)}
```

Reply:
Similarly to the Select, we can allow arbitrary values here and expose a prop to control how it's serialized.

Reply:
What valid use cases are there for something other than a string or number?